DMCompiler/DM/Visitors/DMVisitorExpression.cs

```diff
@@ -564,6 +564,19 @@ public void VisitIdentifier(DMASTIdentifier identifier)
             Result = new Expressions.NewList(newList.Location, expressions);
         }
 
+        public void VisitAddText(DMASTAddText addText) {
+            DMExpression[] exp_arr = new DMExpression[addText.Parameters.Length];
+            if (exp_arr.Length < 2) throw new CompileErrorException(addText.Location, "Invalid addtext() parameter count; expected 2 or more arguments");
```

Comment:
Very minor, but do the parameter check first to prevent an unnecessary allocation.
```suggestion
            if (addText.Parameters.Length < 2) throw new CompileErrorException(addText.Location, "Invalid addtext() parameter count; expected 2 or more arguments");
            DMExpression[] exp_arr = new DMExpression[addText.Parameters.Length];
```

Reply:
It's okay, I too care a ridiculous amount about unnecessary mallocs